Modules/EventHandler.lua

```diff
@@ -6,56 +6,70 @@ local Stats = ECSLoader:ImportModule("Stats")
 function EventHandler.HandleOnEvent(_, event, ...)
-    local currentTime = GetTime()
-    local shouldDebounce = currentTime - lastEventTime < DEBOUNCE_INTERVAL
-    lastEventTime = currentTime
-
     local args = {...}
 
-    if (not shouldDebounce) and event == "GROUP_ROSTER_UPDATE" then
-        -- Someone joined or left the group
-        currentGroupMembers = GetNumGroupMembers()
-    elseif ((not shouldDebounce) and event == "UNIT_AURA")
-            or event == "PLAYER_LEVEL_UP"
-            or event == "CHARACTER_POINTS_CHANGED"
-            or event == "RUNE_UPDATED" then
-        if currentGroupMembers > 5 then
-            -- When in a raid update on the next UPDATE_INTERVAL tick
-            shouldUpdate = true
-        else
-            -- Otherwise update right away
-            Stats.UpdateInformation()
+    if event == "PLAYER_ENTERING_WORLD" then
+        return
+    end
+
+    local statsFrame = Stats:GetFrame()
+
+    if statsFrame:IsVisible() then
+        if event == "ACTIVE_TALENT_GROUP_CHANGED" or
+            event == "CHARACTER_POINTS_CHANGED" or
+            event == "COMBAT_RATING_UPDATE" or
+            event == "PLAYER_DAMAGE_DONE_MODS" or
+            event == "PLAYER_LEVEL_UP" or
+            event == "PLAYER_MOUNT_DISPLAY_CHANGED" or
+            event == "PLAYER_TALENT_UPDATE" or
+            event == "RUNE_UPDATED" or
+            event == "SKILL_LINES_CHANGED" or
+            event == "SPELL_POWER_CHANGED" or
+            event == "UPDATE_SHAPESHIFT_FORM" or ((
+                event == "UNIT_ATTACK" or
+                event == "UNIT_ATTACK_SPEED" or
+                event == "UNIT_AURA" or
+                event == "UNIT_DAMAGE" or
+                event == "UNIT_SPELL_HASTE" or
+                event == "UNIT_STATS" or
+                event == "UNIT_RANGEDDAMAGE" or
+                event == "UNIT_RANGED_ATTACK_POWER"
+            ) and args[1] == "player") then
+            statsFrame:SetScript("OnUpdate", DelayedUpdateInformation)
+        elseif event == "PLAYER_EQUIPMENT_CHANGED" or event == "SOCKET_INFO_SUCCESS" then
+            PaperDollFrame:SetScript("OnUpdate", DelayedUpdateGearColorFrames)
+            statsFrame:SetScript("OnUpdate", DelayedUpdateInformation)
         end
-    elseif event == "PLAYER_EQUIPMENT_CHANGED" or event == "SOCKET_INFO_SUCCESS" then
-        GearInfos.UpdateGearColorFrames()
-        C_Timer.After(0.5, function ()
-            Stats.UpdateInformation()
-        end)
-    elseif event == "PLAYER_MOUNT_DISPLAY_CHANGED" then
-        C_Timer.After(0.5, function ()
-            Stats.UpdateInformation()
-        end)
-    elseif ((not shouldDebounce) and event == "INSPECT_READY" and args[1] == UnitGUID("target")) then
-        GearInfos:UpdateInspectGearColorFrames()
     end
-end
\ No newline at end of file
+    if event == "INSPECT_READY" and args[1] == UnitGUID("target") then
+        InspectPaperDollFrame:SetScript("OnUpdate", DelayedUpdateInspectGearColorFrames)
```

Comment:
I think a simple `nil` check for `InspectPaperDollFrame` is enough to correctly handle the `INSPECT_READY` events fire prematurely by TacoTip (and other addons). From my testing the event is fired once on mouse over/target and then another time when the `InspectPaperDollFrame` is actually opened.